libtsuba/include/tsuba/RDGTopology.h

```diff
@@ -0,0 +1,303 @@
+      uint64_t num_edges, const uint64_t* adj_indices, uint64_t num_nodes,
+      const uint32_t* dests, TopologyKind topology_state,
+      TransposeKind transpose_state, EdgeSortKind edge_sort_state,
+      NodeSortKind node_sort_state);
+
+  /// Make an RDGTopology for an EdgeShuffle related Topology from in memory structures
+  static katana::Result<tsuba::RDGTopology> Make(
+      uint64_t num_edges, const uint64_t* adj_indices, uint64_t num_nodes,
+      const uint32_t* dests, TopologyKind topology_state,
+      TransposeKind transpose_state, EdgeSortKind edge_sort_state,
+      const uint64_t* edge_index_to_property_index_map,
+      const uint8_t* edge_condensed_type_id_map_,
+      const uint8_t* node_condensed_type_id_map_);
+
+  /// Make and fully populate an RDGTopology from in memory structures
+  static katana::Result<tsuba::RDGTopology> Make(
+      uint64_t num_edges, const uint64_t* adj_indices, uint64_t num_nodes,
+      const uint32_t* dests, TopologyKind topology_state,
+      TransposeKind transpose_state, EdgeSortKind edge_sort_state,
+      NodeSortKind node_sort_state,
+      const uint64_t* edge_index_to_property_index_map,
+      const uint64_t* node_index_to_property_index_map,
+      const uint8_t* edge_condensed_type_id_map_,
+      const uint8_t* node_condensed_type_id_map_);
+
+  // Make an RDGTopology from on storage metadata
+  static katana::Result<tsuba::RDGTopology> Make(
+      PartitionTopologyMetadataEntry* entry);
+
+private:
+  //** IMPORTANT** if you add any fields, you must add them to the assignment operator in the cpp file
+  //TODO: emcginnis, this ^ is gross. how can we make this not required?
+
+  // valid immediately, stored in metadata
+  uint64_t num_edges_{0};
+  uint64_t num_nodes_{0};
+  //TODO: emcginnis PartitionMetadata.h has a transposed flag, see if that is the same thing and replace it?
+  TopologyKind topology_state_{-1};
+  TransposeKind transpose_state_{-1};
+  EdgeSortKind edge_sort_state_{-1};
+  NodeSortKind node_sort_state_{-1};
+
+  // File store state
+  /// Flag to show if we have mapped the file store to memory
+  bool file_store_mapped_{false};
+  /// Flag to show if we have bound the file at PartitionTopologyMetadataEntry.path_ to a file store
+  bool file_store_bound_{false};
+  /// Flag to show if the file on disk is up to date with our in memory represenation
+  bool storage_valid_{false};
+
+  // index into the partition_metadata_entries array, also indicates this RDGTopology came from storage
+  // rather than purely created in memory
+  PartitionTopologyMetadataEntry* metadata_entry_{nullptr};
+
+  // must be loaded from file store or set
+  const uint64_t* adj_indices_{nullptr};
+  const uint32_t* dests_{nullptr};
+  const uint64_t* edge_index_to_property_index_map_{nullptr};
+  const uint64_t* node_index_to_property_index_map_{nullptr};
+  const uint8_t* edge_condensed_type_id_map_{nullptr};
+  const uint8_t* node_condensed_type_id_map_{nullptr};
```

Comment:
What is a condensed type id map?

Reply:
Your description is correct.

Reply:
From talking with Amber, it seems `GraphTopologyTypes::Entitytype` came about before `katana::EntityTypeID` was fully realized.
Going to convert to using `katana::EntityTypeID`